doc/sources/algorithms.rst

```diff
@@ -20,177 +20,331 @@
-     - PCA
-     - All parameters except ``svd_solver`` != 'full'.
-     - Sparse data is not supported.
-   * - Dimensionality reduction
-     - TSNE
-     - All parameters except ``metric`` != 'euclidean' or 'minkowski' with ``p`` != 2.
-     - Sparse data is not supported.
-   * - Unsupervised
-     - NearestNeighbors
+     - No limitations
+   * - `RandomForestRegressor`
+     - All parameters are supported except:
+
+       - ``warm_start`` = `True`
+       - ``cpp_alpha`` != `0`
+       - ``criterion`` != `'mse'`
+     - Multi-output and sparse data are not supported
+   * - `KNeighborsRegressor`
+     - All parameters are supported except:
+
+       - ``metric`` != `'euclidean'` or `'minkowski'` with ``p`` != `2`
+     - Multi-output and sparse data are not supported
+   * - `LinearRegression`
+     - All parameters are supported except:
+
+       - ``normalize`` != `False`
+       - ``sample_weight`` != `None`
+     - Only dense data is supported, `#observations` should be >= `#features`.
+   * - `Ridge`
+     - All parameters are supported except:
+
+       - ``normalize`` != `False`
+       - ``solver`` != `'auto'`
+       - ``sample_weight`` != `None`
+     - Only dense data is supported, `#observations` should be >= `#features`.
+   * - `ElasticNet`
+     - All parameters are supported except:
+
+       - ``sample_weight`` != `None`
+     - Multi-output and sparse data are not supported, `#observations` should be >= `#features`.
+   * - `Lasso`
+     - All parameters are supported except:
+
+       - ``sample_weight`` != `None`
+     - Multi-output and sparse data are not supported, `#observations` should be >= `#features`.
+
+Clustering
+**********
+
+.. list-table::
+   :widths: 10 30 20
+   :header-rows: 1
+   :align: left
+
+   * - Algorithm
+     - Parameters
+     - Data formats
+   * - `KMeans`
+     - All parameters are supported except:
+
+       - ``precompute_distances``
+       - ``sample_weight`` != `None`
+     - No limitations
+   * - `DBSCAN`
+     - All parameters are supported except:
+
+       - ``metric`` != `'euclidean'` or `'minkowski'` with ``p`` != `2`
+       - ``algorithm`` != `'brute'` or `'auto'`
+     - Only dense data is supported
+
+Dimensionality reduction
+************************
+
+.. list-table::
+   :widths: 10 30 20
+   :header-rows: 1
+   :align: left
+
+   * - Algorithm
+     - Parameters
+     - Data formats
+   * - `PCA`
+     - All parameters are supported except:
+
+       - ``svd_solver`` != `'full'`
+     - Sparse data is not supported
+   * - `TSNE`
+     - All parameters are supported except:
+
+       - ``metric`` != 'euclidean' or `'minkowski'` with ``p`` != `2`
+
+       Refer to :ref:`TSNE acceleration details <acceleration_tsne>` to learn more.
+     - Sparse data is not supported
+
+Other tasks
+***********
+
+.. list-table::
+   :widths: 10 30 20
+   :header-rows: 1
+   :align: left
+
+   * - Algorithm
+     - Parameters
+     - Data formats
+   * - `NearestNeighbors`
```

Comment:
"Other tasks" -> "Unsupervised learning" for NearestNeighbors?
https://scikit-learn.org/stable/modules/neighbors.html#unsupervised-nearest-neighbors

Reply:
clustering and dimensionality reduction would also be considered unsupervised learning, and so it doesn't make sense to have unsupervised learning as a separate category of tasks if we already have clustering and dimensionality reduction categories.

should there be a separate category for this algorithm? "Nearest Neighbors", the same wat we have in oneDAL docs:

![image](https://user-images.githubusercontent.com/17865025/158989899-3d0c8d92-1923-4264-803e-c2a8c6b5f292.png)

Reply:
@KalyanovD  do you agree with the suggestion?